src/mumble/ListenerLocalVolumeSlider.cpp

```diff
@@ -0,0 +1,37 @@
+// Copyright 2005-2020 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "ListenerLocalVolumeSlider.h"
+
+#include <cmath>
+
+// We define a global macro called 'g'. This can lead to issues when included code uses 'g' as a type or parameter name
+// (like protobuf 3.7 does). As such, for now, we have to make this our last include.
+#include "Global.h"
+
+ListenerLocalVolumeSlider::ListenerLocalVolumeSlider(ClientUser *user, Channel *channel, QWidget *parent)
+	: VolumeSliderWidgetAction(parent), m_user(user), m_channel(channel) {
```

Comment:
What is the user needed for?

Also the channel should be const.

And probably using references instead of pointers would be better. We don't want to allow `nullptr`as an argument here anyways, do we?

Reply:
this was mostly mimicking the current ListenerLocalVolumeDialogs implementation which also has the user and doesn't use it for anything and takes the channel as a pointer. 

I am checking for `nullptr` where needed. If requested I can change it to references.

Reply:
> his was mostly mimicking the current ListenerLocalVolumeDialogs implementation which also has the user and doesn't use it for anything and takes the channel as a pointer.

Well I guess then that class has to be reworked at some point as well ^^

> I am checking for nullptr where needed. 

Yes, but still the class doesn't really make sense if the channel is `nullptr` since essentially that'd cancel all functionality. Thus `nullptr` is effectively not a valid parameter that can be passed to the class.
And given that the channel is not optional but mandatory, we should make that semantically clear by using references instead.

> If requested I can change it to references.

Yes, please.
I know this is currently not done in like 80-90% of the cases but we are striving towards improved code quality and things like these help with that :)